modules/1plusXRtdProvider.js

```diff
@@ -238,15 +233,12 @@ export const setTargetingDataToConfig = (papiResponse, { bidders }) => {
     const ortb2Updates = buildOrtb2Updates({ segments, topics }, bidder);
     const updatedBidderConfig = updateBidderConfig(bidder, ortb2Updates, bidderConfigs);
     if (updatedBidderConfig) {
+      setAppnexusAudiences(segments);
       config.setBidderConfig({
```

Comment:
since version 7 this is not the correct way to add first party data, see [this for some more info](https://docs.prebid.org/dev-docs/add-rtd-submodule.html#getbidrequestdata) (including the yellow warning about this pattern specifically).

The first argument to `getBidRequestData` contains an `ortb2Fragments` object that you can augment, the logic will stay very similar but instead of doing `setBidderConfig({config: {ortb2: myUpdatedFPD}})` it should do `mergeDeep(ortb2Fragments.bidder[bidderCode], myUpdatedFPD)` for each bidderCode you want to pass data to; or merge into ortb2Fragments.global directly if that's appropriate.

The issue with using `setBidderConfig` is that it won't take effect until the next auction (since v7).